templates/mod/ssl.conf.erb

```diff
@@ -12,7 +12,9 @@
   SSLSessionCacheTimeout <%= @ssl_sessioncachetimeout %>
   <%- if scope.function_versioncmp([@_apache_version, '2.4']) >= 0 -%>
   Mutex <%= @_ssl_mutex %>
+    <%- if @ssl_compression -%>
```

Comment:
and that's exactly what I wanted, and wrote the spec tests for it ;)

I'm on OpenBSD*, and there Apache 2.4 from packages is built against libressl, which doesn't support SSLCompression at all. 

Therefore, when the module sets 
SSLCompression off

I get this error message on apache startup, and it refuses to start:

H00526: Syntax error on line 14 of /etc/apache2/modules/ssl.conf:
Setting Compression mode unsupported; not implemented by the SSL library

The default value for SSLCompression, if not given in the configuration file, is 'off', see:
https://httpd.apache.org/docs/2.4/mod/mod_ssl.html#sslcompression

The puppet module defaults the $ssl_compression parameter of the class to false, following the 
Apache default value, which is fine. But since it's the default value, it is not necessary to explicitly 
set it to 'off'. That said, for every Apache/SSL combination, that supports SSLCompression, nothing changes, for Apache/SSL combination, that doesn't support SSLCompression, things are not broken anymore (:
People that want to have SSLCompression, have to enable it explicitly in any case anyways, and likely know if their Apache/SSL combination supports it. For those the statement will be added to the config file as it was done before.

Does that make sense?

* yes, yes, OpenBSD is not (yet ;) supported by the module. But there may be other OS/Distros that run into the same issue...